### PR TITLE
Docker: solve polkadot relay folder renaming and other small fixes

### PR DESCRIPTION
**Pull request by gpmayorga**
Includes:
- Polkadot relays now expect their folder to be named "polkadot" rather than "relay-chain". The new docker container is made aware of this and will rename the folder if needed.
- Dockerfile build based on Parity "rust-ready" production base image
- More lean Dockerfile

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@ updates:
   - package-ecosystem: "github-actions"
     directory: "/"
     schedule:
-      interval: "daily"
+      interval: "monthly"
```
Comment (wischli) [inline .github/dependabot.yml:7]: 🚀

File: .github/workflows/build-docker.yml [hunk 2]
```diff
@@ -51,10 +40,10 @@ jobs:
...
           tags: |
             type=semver,pattern={{raw}},prefix=${{ matrix.target == 'test' && 'test-' || '' }}
-            type=edge,event=pr,suffix={{sha}}-${{ env.NOW }},prefix=${{ matrix.target == 'test' && 'test-' || '' }}
-            type=ref,event=tag,suffix={{sha}}-${{ env.NOW }},prefix=${{ matrix.target == 'test' && 'test-' || '' }}
-            type=ref,event=pr,suffix={{sha}}-${{ env.NOW }},prefix=${{ matrix.target == 'test' && 'test-' || '' }}
-            type=ref,event=branch,prefix=${{ matrix.target == 'test' && 'test-' || '' }}-{{branch}},suffix={{sha}}-${{ env.NOW }}
+            type=edge,event=pr,suffix=-{{sha}}-${{ env.NOW }},prefix=${{ matrix.target == 'test' && 'test-' || '' }}
+            type=ref,event=tag,suffix=-{{sha}}-${{ env.NOW }},prefix=${{ matrix.target == 'test' && 'test-' || '' }}
+            type=ref,event=pr,suffix=-{{sha}}-${{ env.NOW }},prefix=${{ matrix.target == 'test' && 'test-' || '' }}PR
+            type=ref,event=branch,prefix=${{ matrix.target == 'test' && 'test-' || '' }},suffix=-{{sha}}-${{ env.NOW }}
```
Comment (wischli) [inline .github/workflows/build-docker.yml:46]: One community member proposed having a dedicated `latest` tag for the latest releases. The advantage would be staying up-to-date without needing to change the docker tag. Downside potential issues due to breaking changes.

AFAICT, we don't have this at the moment right? WDYT @gpmayorga? If we decide for it, it does not have to be part of this PR!

File: docker/scripts/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+if [ "$1" == "--help" ]; then
+    echo "No arguments detected, printing help and exiting..."
+    centrifuge-chain "$@"
+    exit 0
+fi
+
+# Fix to account for Polkadot's renaming of their DB folder from
+# relay-chain to polkadot. Probably not needed after all nodes are upgraded
+# beyond Polkadot 0.93.z
```
Comment (wischli) [inline docker/scripts/entrypoint.sh:10]:
```suggestion
# beyond Polkadot 0.9.42+
```

**Review by wischli — APPROVED** [review]
Thanks for fixing this! 🤞 switching to Parity as docker base image won't bite us long term

**Comment by gpmayorga** [comment]
Some highlights 

on the Docker changes we merged this last two weeks:
Security vulnerabilities went from 10 High and 200 Medium to 0 High and 5 Medium
Screenshot 2023-11-28 at 14 36 43
Screenshot 2023-11-28 at 14 36 30

Image 20% smaller.
Screenshot 2023-11-28 at 14 36 46
Screenshot 2023-11-28 at 14 36 32

Better overall security by not using root everywhere
Very aligned with the guidelines from parity and generally running chain nodes in Docker containers

**Comment by gpmayorga** [comment]
@wischli the PR is ready for prime time, I opened a couple of discussions for some decisions I want to double-check but this should be a pretty solid Docker strategy at this point

**Review by mustermeiszer — COMMENTED** [review]
From my limit knowledge this looks really good! 

Will approve once somebody with more knowledge cann second my opinion ^^

**Pull request merged**